src/services/importService.ts

```diff
@@ -0,0 +1,80 @@
+import { Collection } from 'types/collection';
+import { ElectronFile, FileWithCollection } from 'types/upload';
+import { runningInBrowser } from 'utils/common';
+import { getElectronFiles } from 'utils/upload';
+
+class ImportService {
+    ElectronAPIs: any;
+    private allElectronAPIsExist: boolean = false;
+
+    constructor() {
+        this.ElectronAPIs = runningInBrowser() && window['ElectronAPIs'];
+        this.allElectronAPIsExist = !!this.ElectronAPIs?.exists;
+    }
+
+    async setDoneUploadingFiles() {
+        if (this.allElectronAPIsExist) {
+            this.ElectronAPIs.setToUploadFiles(null, null, null, true);
+        }
+    }
+
+    async getIfToUploadFilesExists(): Promise<boolean> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.getIfToUploadFilesExists();
+        }
+    }
+
+    async getElectronFile(filePath: string): Promise<ElectronFile> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.getElectronFile(filePath);
+        }
+    }
+
+    async showUploadFilesDialog(): Promise<string[]> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.showUploadFilesDialog();
+        }
+    }
+
+    async showUploadDirsDialog(): Promise<string[]> {
+        if (this.allElectronAPIsExist) {
+            return this.ElectronAPIs.showUploadDirsDialog();
+        }
+    }
+
+    async getToUploadFiles() {
+        if (this.allElectronAPIsExist) {
+            const { filesPaths, collectionName, collectionIDs } =
+                this.ElectronAPIs.getToUploadFiles();
+            const files = await getElectronFiles(filesPaths);
+            return {
+                files,
+                collectionName,
+                collectionIDs,
+            };
+        }
+    }
+
+    async setToUploadFiles(
+        files: FileWithCollection[],
+        collections?: Collection[]
+    ) {
+        if (this.allElectronAPIsExist) {
+            let collectionName;
+            if (collections?.length > 0) {
+                collectionName = collections[0].name;
```

Comment:
we can just store the file paths and the  in case the user chose to create a new collection with a custom name we should store that  `collectionName` , everything else can be derived na

Reply:
I think we still need `collectionIDs`. Because we don't have the collection name if the user chose to upload to some existing collection as then this `collections` array would be empty. I couldn't find a way to find out the collection name except for storing the IDs. Should we have some state for `collectionName` that would get updated (or is there already something)?

Reply:
I have removed the part of storing `collectionIDs`, now it will store `collectionName` in both cases of an existing or a new collection.

Reply:
Yep, it did worked.